vulnerabilities/importers/pysec.py

```diff
@@ -0,0 +1,219 @@
+        """
+        1. Fetch the data from osv api
+        2. unzip the file
+        3. open the file one by one
+        4. yield the json file to parse_advisory_data
+        """
+        url = "https://osv-vulnerabilities.storage.googleapis.com/PyPI/all.zip"
+        try:
+            response = requests.get(url).content
+            with ZipFile(BytesIO(response)) as zip_file:
+                for file_name in zip_file.namelist():
+                    with zip_file.open(file_name) as f:
+                        vul_info = json.loads(f.read())
+                        yield parse_advisory_data(vul_info)
+        except requests.exceptions.RequestException:
+            logger.error("Failed to fetch osv-vulnerabilities PyPI")
+
+
+def parse_advisory_data(raw_data: dict) -> AdvisoryData:
+    summary = raw_data.get("summary") or ""
+    aliases = get_aliases(raw_data)
+    date_published = get_published_date(raw_data)
+    severity = get_severity(raw_data)
+    references = get_references(raw_data, severity)
+
+    affected_package = []
+    if "affected" in raw_data:
+        for affected_pkg in raw_data["affected"]:
+            purl = get_aff_purl(affected_pkg, raw_data["id"])
+            affected_version_range = get_aff_version_range(affected_pkg, raw_data["id"], purl)
+
+            for fixed_range in affected_pkg.get("ranges", []):
+                fixed_version = get_fixed_version(fixed_range, raw_data["id"], purl)
+
+                for version in fixed_version:
+                    affected_package.append(
+                        AffectedPackage(
+                            package=purl,
+                            affected_version_range=affected_version_range,
+                            fixed_version=version,
+                        )
+                    )
+    if not "affected" in raw_data:
+        logger.error(f"affected_package not found - {raw_data['id'] !r}")
+
+    return AdvisoryData(
+        aliases=aliases,
+        summary=summary,
+        affected_packages=affected_package,
+        references=references,
+        date_published=date_published,
+    )
+
+
+def fixed_filter(fixed_range) -> []:
+    """
+    >>> fixed_filter({"type": "SEMVER", "events": [{"introduced": "0"}, {"fixed": "1.6.0"}]})
+    ['1.6.0']
+    >>> fixed_filter({"type": "ECOSYSTEM","events":[{"introduced": "0"},{"fixed": "1.0.0"},{"fixed": "9.0.0"}]})
+    ['1.0.0', '9.0.0']
+    """
+    filter_fixed = list(filter(lambda x: x.keys() == {"fixed"}, fixed_range["events"]))
+    list_fixed = [i["fixed"] for i in filter_fixed]
+    return list_fixed
+
+
+def get_aliases(raw_data) -> []:
+    """
+    aliases field is optional , id is required and these are all aliases from our perspective
+    converting list of two fields to a dict then , convert it to a list to make sure a list is unique
+    """
+    vulnerability_id = raw_data.get("id")
+    vulnerability_aliases = raw_data.get("aliases") or []
```

Comment:
@Hritik14 FWIW, I kinda like `raw_data.get("aliases") or []` as this will gte you a list for sure.
`raw_data.get("aliases", [])` can get you a `None` if `aliases` is `None`, or an empty string or an empty dict or any emptyish value that is set.